rpcs3/Emu/CPU/CPUThread.cpp

```diff
@@ -766,9 +765,9 @@ bool cpu_thread::check_state() noexcept
 			}
 			else
 			{
-				if (cpu_can_stop && !(flags & cpu_flag::wait))
```

Comment:
This block of code was not correct and could have set wait flag on emulation stopping even if we are inside of a function and need wait flag disabled (this can degrade savestates' accuracy for example). I've previously added wait flag after cpu_task() return which is the correct place for it.

Reply:
It also used to uset wait flag on function return (cpu_flag::ret) which is incorrect either.